docs/source/routing/configuration/yaml.mdx

```diff
@@ -0,0 +1,818 @@
+import Batching from '../../../shared/config/batching.mdx';
+import Connectors from '../../../shared/config/connectors.mdx';
+import Coproc from '../../../shared/config/coprocessor.mdx';
+import Cors from '../../../shared/config/cors.mdx';
+import Csrf from '../../../shared/config/csrf.mdx';
+import DemandCtrl from '../../../shared/config/demand_control.mdx';
+import ExperimentalChaos from '../../../shared/config/experimental_chaos.mdx';
+import ExperimentalType from '../../../shared/config/experimental_type_conditioned_fetching.mdx';
+import FleetDetector from '../../../shared/config/fleet_detector.mdx';
+import ForbidMut from '../../../shared/config/forbid_mutations.mdx';
+import Headers from '../../../shared/config/headers.mdx';
+import HealthChk from '../../../shared/config/health_check.mdx';
+import Homepage from '../../../shared/config/homepage.mdx';
+import IncludeSubErr from '../../../shared/config/include_subgraph_errors.mdx';
+import LicenseEnf from '../../../shared/config/license_enforcement.mdx';
+import Limits from '../../../shared/config/limits.mdx';
+import OverrideSubUrl from '../../../shared/config/override_subgraph_url.mdx';
+import PersistedQueries from '../../../shared/config/persisted_queries.mdx';
+import Plugins from '../../../shared/config/plugins.mdx';
+import PreviewEntityCache from '../../../shared/config/preview_entity_cache.mdx';
+import PreviewFileUploads from '../../../shared/config/preview_file_uploads.mdx';
+import ProgOverride from '../../../shared/config/progressive_override.mdx';
+import Rhai from '../../../shared/config/rhai.mdx';
+import Sandbox from '../../../shared/config/sandbox.mdx';
+import Subscription from '../../../shared/config/subscription.mdx';
+import Supergraph from '../../../shared/config/supergraph.mdx';
+import Telemetry from '../../../shared/config/telemetry.mdx';
+import Tls from '../../../shared/config/tls.mdx';
+import TrafficShaping from '../../../shared/config/traffic_shaping.mdx';
+
+This reference covers the YAML configuration file properties for configuring an Apollo Router. 
+
+## YAML configuration properties
+
+The router can be configured by a YAML configuration file. This file enables you to declaratively configure various runtime properties of your router's behavior.
+
+At startup, you set the config file for your router by providing its path with the [`--config`](#-c----config) option: 
+
+```bash
+./router --config router.yaml
+```
+
+<Tip>
+
+Enable your text editor to validate the format and content of your router YAML configuration file by [configuring it with the router's configuration schema](/graphos/routing/configuration/cli#configuration-schema-for-ide-validation).
+
+</Tip>
+
+## Example YAML with all properties
+
+Expand the code block to view an example YAML config file containing all properties.
+
+<ExpansionPanel title="Example router YAML config file with all properties">
+
+<RouterYaml/>
+
+</ExpansionPanel>
+
+## Properties
```

Comment:
From design crit:
For increased discoverability, consider organizing configs by features (Security, Performance, etc.) rather than alphabetically